extensions/smallrye-reactive-messaging/runtime/src/main/java/io/quarkus/smallrye/reactivemessaging/runtime/SmallRyeReactiveMessagingLifecycle.java

```diff
@@ -17,12 +14,9 @@
     MediatorManager mediatorManager;
 
     void onApplicationStart(@Observes StartupEvent event) {
-        CompletableFuture<Void> future = mediatorManager.initializeAndRun();
```

Comment:
Hm, I don't understand why this change? StreamRegistars are initialized asynchronously so we should block and wait until the registration is finished. Otherwise our app might start before everything is ready... or?

Reply:
Actually, it was wrong. The async connection must happen during the actual subscription and not during the initialization. So during the initialization, the connector should not do anything "network related". The code changed in SmallRye Reactive Messaging.

Reply:
(the new method return void and not a CompletionStage)

Reply:
Hm, should the `MediatorManager` method be renamed to `initialize()`?

Reply:
It _triggers_ the subscriptions, so _run_ still make sense. However, you don't control what happens during the subscriptions.

Reply:
Ok. I should supplement my knowledge in this area. I'm still a newbie here ;-)